core/templates/dev/head/domain/exploration/QuestionListObjectFactory.js

```diff
@@ -35,6 +35,14 @@ oppia.factory('QuestionListObjectFactory', [function() {
     this._questions.push(question);
   };
 
+  QuestionList.prototype.removeQuestion = function(question){
+    var index = this._questions.indexOf(question);
+    if(index != -1){
+      return this._questions.splice(index,1);
+    }
+    return ;
```

Comment:
If the error is unexpected in normal operation, just raise it. No need to catch, since its presence means that something's gone wrong and it's something we want to know about.